[Reporting/Visualization] Migrate Visualize to V2 reporting

jloleysens opened the pull request:
## Summary

Updates Visualize to use the new v2 report (powered by locators) that ensures all state can be migrated and as a bonus removes the requirement to first save changes.

New export types were introduced in https://github.com/elastic/kibana/pull/101048.


## How to test

1. Go "Visualize Library"
2. Create a new visualization, one that uses "app/visualize". This includes "Custom visualiztions" powered by Vega. Installing a sample data set should get you some of these
3. Edit the visualization in some way, press "Update"
4. Generate a PDF via "Share" menu, this should now be clickable without first needing to Save the visualization.
5. Do steps 3 & 4 but for an existing visualisation
6. In both cases you should get a report (PDF/PNG) with the latest changes you updated to

## Release note

You can now generate a PDF or PNG report of an unsaved visualization.

<img width="385" alt="Screenshot 2021-08-26 at 13 35 15" src="https://user-images.githubusercontent.com/8155004/130955909-9cfb8625-6ffa-4e60-80fb-454440afafe1.png">

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

ppisljar commented on src/plugins/visualize/public/locator.ts:
```diff
@@ -0,0 +1,92 @@
+
+  uiState?: SerializableRecord;
+
+  vis?: SerializableRecord;
```
`PureVisState` ?

ppisljar commented on src/plugins/visualize/public/locator.ts:
```diff
@@ -0,0 +1,92 @@
+/*
```
should be moved to common and registered on client and server

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/)
* Commit: fcc34f0df1ffd36ff0952dc0d288c9801a4c0274
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/execution/node/412/log)

### History
* :broken_heart: [Build #148862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/) failed 7558ab8aae4de6d98c8f67bef7488c8efd3bfd07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/","number":"148862","commit":"7558ab8aae4de6d98c8f67bef7488c8efd3bfd07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/","number":"149445","commit":"fcc34f0df1ffd36ff0952dc0d288c9801a4c0274"}]}
PIPELINE-->

jloleysens commented:
Thanks for the review @ppisljar , I think I've addressed your points. Do you think you could take another look?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/)
* Commit: 57d7cad693b86d2b7522feb2fb8917282994b2ec
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110206/57d7cad693b86d2b7522feb2fb8917282994b2ec)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup2](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/execution/node/604/log)
* [Execute xpack-kibana-ciGroup2](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/execution/node/1063/log)

### History
* :broken_heart: [Build #149445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/) failed fcc34f0df1ffd36ff0952dc0d288c9801a4c0274
* :broken_heart: [Build #148862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/) failed 7558ab8aae4de6d98c8f67bef7488c8efd3bfd07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/","number":"148862","commit":"7558ab8aae4de6d98c8f67bef7488c8efd3bfd07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/","number":"149445","commit":"fcc34f0df1ffd36ff0952dc0d288c9801a4c0274"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/","number":"149510","commit":"57d7cad693b86d2b7522feb2fb8917282994b2ec"}]}
PIPELINE-->

ppisljar reviewed (approved):
code LGTM

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/)
* Commit: 45e46d1e526e940f75bf354dfb41fd94939be1c8
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110206/45e46d1e526e940f75bf354dfb41fd94939be1c8)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/b7ad4268d0ecc95dbbb10e6a3de06f4bd7de635c) | [after](https://github.com/elastic/kibana/commit/45e46d1e526e940f75bf354dfb41fd94939be1c8) | diff |
| --- | --- | --- | --- |
| `visualize` | 81 | 83 | +2 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/b7ad4268d0ecc95dbbb10e6a3de06f4bd7de635c) | [after](https://github.com/elastic/kibana/commit/45e46d1e526e940f75bf354dfb41fd94939be1c8) | diff |
| --- | --- | --- | --- |
| `visualizations` | 261 | 253 | -8 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b7ad4268d0ecc95dbbb10e6a3de06f4bd7de635c) | [after](https://github.com/elastic/kibana/commit/45e46d1e526e940f75bf354dfb41fd94939be1c8) | diff |
| --- | --- | --- | --- |
| `visualize` | 95.4KB | 96.2KB | +792.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b7ad4268d0ecc95dbbb10e6a3de06f4bd7de635c) | [after](https://github.com/elastic/kibana/commit/45e46d1e526e940f75bf354dfb41fd94939be1c8) | diff |
| --- | --- | --- | --- |
| `visualize` | 27.8KB | 31.0KB | +3.2KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/b7ad4268d0ecc95dbbb10e6a3de06f4bd7de635c) | [after](https://github.com/elastic/kibana/commit/45e46d1e526e940f75bf354dfb41fd94939be1c8) | diff |
| --- | --- | --- | --- |
| `visualizations` | 279 | 271 | -8 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/b7ad4268d0ecc95dbbb10e6a3de06f4bd7de635c) | [after](https://github.com/elastic/kibana/commit/45e46d1e526e940f75bf354dfb41fd94939be1c8) | diff |
| --- | --- | --- | --- |
| `visualize` | 74 | 89 | +15 |


</details>

### History
* :broken_heart: [Build #149510](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/) failed 57d7cad693b86d2b7522feb2fb8917282994b2ec
* :broken_heart: [Build #149445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/) failed fcc34f0df1ffd36ff0952dc0d288c9801a4c0274
* :broken_heart: [Build #148862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/) failed 7558ab8aae4de6d98c8f67bef7488c8efd3bfd07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/","number":"148862","commit":"7558ab8aae4de6d98c8f67bef7488c8efd3bfd07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/","number":"149445","commit":"fcc34f0df1ffd36ff0952dc0d288c9801a4c0274"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/","number":"149510","commit":"57d7cad693b86d2b7522feb2fb8917282994b2ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/","number":"149692","commit":"45e46d1e526e940f75bf354dfb41fd94939be1c8"}]}
PIPELINE-->

stratoula reviewed (commented):
@jloleysens this PR looks great!
I see that there are some sample viz that fail. For example:
- TSVB 
![image](https://user-images.githubusercontent.com/17003240/131506188-8a38f7cc-c177-48e1-a573-6d19656fce5d.png)
- Vertical bar
![image](https://user-images.githubusercontent.com/17003240/131506431-4ad6d30b-e128-4fa2-9b79-5bcb37fd6e6f.png)
- Vega
![image](https://user-images.githubusercontent.com/17003240/131506560-c722ad99-9617-4feb-8395-d3e07b3ddcfd.png)

I also tried to create a new area viz and generate the report without saving it first, it failed

stratoula commented:
@elasticmachine merge upstream

jloleysens commented:
Great chat @stratoula 🍻 :! I had not considered the encoding issue after my most recent push. I updated the code (and the tests). Do you think you could take another look?

stratoula commented:
@jloleysens thank you so much ❤️ It is quite late for me now, I will review it first thing tomorrow morning!

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/)
* Commit: 760e7028c2dfa3a7fd14221d7d7f0722fe6f2858
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110206/760e7028c2dfa3a7fd14221d7d7f0722fe6f2858)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/760e7028c2dfa3a7fd14221d7d7f0722fe6f2858) | diff |
| --- | --- | --- | --- |
| `visualize` | 81 | 83 | +2 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/760e7028c2dfa3a7fd14221d7d7f0722fe6f2858) | diff |
| --- | --- | --- | --- |
| `visualizations` | 261 | 253 | -8 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/760e7028c2dfa3a7fd14221d7d7f0722fe6f2858) | diff |
| --- | --- | --- | --- |
| `visualize` | 95.4KB | 96.2KB | +792.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/760e7028c2dfa3a7fd14221d7d7f0722fe6f2858) | diff |
| --- | --- | --- | --- |
| `visualize` | 27.8KB | 31.0KB | +3.2KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/760e7028c2dfa3a7fd14221d7d7f0722fe6f2858) | diff |
| --- | --- | --- | --- |
| `visualizations` | 279 | 271 | -8 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/760e7028c2dfa3a7fd14221d7d7f0722fe6f2858) | diff |
| --- | --- | --- | --- |
| `visualize` | 74 | 89 | +15 |


</details>

### History
* :green_heart: [Build #149692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/) succeeded 45e46d1e526e940f75bf354dfb41fd94939be1c8
* :broken_heart: [Build #149510](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/) failed 57d7cad693b86d2b7522feb2fb8917282994b2ec
* :broken_heart: [Build #149445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/) failed fcc34f0df1ffd36ff0952dc0d288c9801a4c0274
* :broken_heart: [Build #148862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/) failed 7558ab8aae4de6d98c8f67bef7488c8efd3bfd07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/","number":"148862","commit":"7558ab8aae4de6d98c8f67bef7488c8efd3bfd07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/","number":"149445","commit":"fcc34f0df1ffd36ff0952dc0d288c9801a4c0274"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/","number":"149510","commit":"57d7cad693b86d2b7522feb2fb8917282994b2ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/","number":"149692","commit":"45e46d1e526e940f75bf354dfb41fd94939be1c8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/","number":"149792","commit":"760e7028c2dfa3a7fd14221d7d7f0722fe6f2858"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/)
* Commit: c8e03e1512325e3a9f988a26e0b58862bc86d1ef
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110206/c8e03e1512325e3a9f988a26e0b58862bc86d1ef)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/c8e03e1512325e3a9f988a26e0b58862bc86d1ef) | diff |
| --- | --- | --- | --- |
| `visualize` | 81 | 83 | +2 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/c8e03e1512325e3a9f988a26e0b58862bc86d1ef) | diff |
| --- | --- | --- | --- |
| `visualizations` | 261 | 253 | -8 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/c8e03e1512325e3a9f988a26e0b58862bc86d1ef) | diff |
| --- | --- | --- | --- |
| `visualize` | 95.4KB | 88.1KB | -7.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/c8e03e1512325e3a9f988a26e0b58862bc86d1ef) | diff |
| --- | --- | --- | --- |
| `visualize` | 27.8KB | 39.5KB | +11.6KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/c8e03e1512325e3a9f988a26e0b58862bc86d1ef) | diff |
| --- | --- | --- | --- |
| `visualizations` | 279 | 271 | -8 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/c510288c05089699a897e43c34b28c319ce9a472) | [after](https://github.com/elastic/kibana/commit/c8e03e1512325e3a9f988a26e0b58862bc86d1ef) | diff |
| --- | --- | --- | --- |
| `visualize` | 74 | 89 | +15 |


</details>

### History
* :green_heart: [Build #149792](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/) succeeded 760e7028c2dfa3a7fd14221d7d7f0722fe6f2858
* :green_heart: [Build #149692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/) succeeded 45e46d1e526e940f75bf354dfb41fd94939be1c8
* :broken_heart: [Build #149510](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/) failed 57d7cad693b86d2b7522feb2fb8917282994b2ec
* :broken_heart: [Build #149445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/) failed fcc34f0df1ffd36ff0952dc0d288c9801a4c0274
* :broken_heart: [Build #148862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/) failed 7558ab8aae4de6d98c8f67bef7488c8efd3bfd07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148862/","number":"148862","commit":"7558ab8aae4de6d98c8f67bef7488c8efd3bfd07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/","number":"149445","commit":"fcc34f0df1ffd36ff0952dc0d288c9801a4c0274"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/","number":"149510","commit":"57d7cad693b86d2b7522feb2fb8917282994b2ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/","number":"149692","commit":"45e46d1e526e940f75bf354dfb41fd94939be1c8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/","number":"149792","commit":"760e7028c2dfa3a7fd14221d7d7f0722fe6f2858"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/","number":"149824","commit":"c8e03e1512325e3a9f988a26e0b58862bc86d1ef"}]}
PIPELINE-->

stratoula commented:
Nice, now I don't see the errors but I can see something strange. I have this tagcloud viz
![image](https://user-images.githubusercontent.com/17003240/131622202-8cbdbe4d-e3f8-46e2-8b74-4fc451979f04.png)
but the report is
![image](https://user-images.githubusercontent.com/17003240/131622264-dbc32b42-c78e-4466-846d-261902bdd67d.png)

some tags are missing. I suspect something goes wrong with the time range?

Moreover when I create a new visualization, don't save it and generate the report, I get warnings and the result is the listing page
![image](https://user-images.githubusercontent.com/17003240/131622579-a7f62e17-6a5e-4446-a405-74e64f033c56.png)

jloleysens commented:
@elasticmachine merge upstream

jloleysens commented:
Thanks a lot for the testing and review so far @stratoula !

> Nice, now I don't see the errors but I can see something strange. I have this tagcloud viz

Yeah, there was a TODO to add time range, so I added it (b8ef084) :D

> Moreover when I create a new visualization, don't save it and generate the report, I get warnings and the result is the listing page

Nice, it looks like this was because the `savedSearchId` was not being passed through to reporting. I've updated this.

If you look in `src/plugins/visualize/public/application/utils/get_top_nav_config.tsx`, is there any other param that Visualize might be looking for that's missing from `const params: VisualizeLocatorParams = {...`?

Otherwise I think I've addressed the issues you found. Would you mind taking another look?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/)
* Commit: a582d0ed6b30ceab32f85de1f92932392ba00496
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110206/a582d0ed6b30ceab32f85de1f92932392ba00496)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/b77ca9392b659eac9836d0e3bbe17f333b95ee68) | [after](https://github.com/elastic/kibana/commit/a582d0ed6b30ceab32f85de1f92932392ba00496) | diff |
| --- | --- | --- | --- |
| `visualize` | 81 | 83 | +2 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/b77ca9392b659eac9836d0e3bbe17f333b95ee68) | [after](https://github.com/elastic/kibana/commit/a582d0ed6b30ceab32f85de1f92932392ba00496) | diff |
| --- | --- | --- | --- |
| `visualizations` | 261 | 253 | -8 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b77ca9392b659eac9836d0e3bbe17f333b95ee68) | [after](https://github.com/elastic/kibana/commit/a582d0ed6b30ceab32f85de1f92932392ba00496) | diff |
| --- | --- | --- | --- |
| `visualize` | 95.4KB | 88.1KB | -7.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b77ca9392b659eac9836d0e3bbe17f333b95ee68) | [after](https://github.com/elastic/kibana/commit/a582d0ed6b30ceab32f85de1f92932392ba00496) | diff |
| --- | --- | --- | --- |
| `visualize` | 27.8KB | 39.5KB | +11.6KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/b77ca9392b659eac9836d0e3bbe17f333b95ee68) | [after](https://github.com/elastic/kibana/commit/a582d0ed6b30ceab32f85de1f92932392ba00496) | diff |
| --- | --- | --- | --- |
| `visualizations` | 279 | 271 | -8 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/b77ca9392b659eac9836d0e3bbe17f333b95ee68) | [after](https://github.com/elastic/kibana/commit/a582d0ed6b30ceab32f85de1f92932392ba00496) | diff |
| --- | --- | --- | --- |
| `visualize` | 78 | 93 | +15 |


</details>

### History
* :green_heart: [Build #149824](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/) succeeded c8e03e1512325e3a9f988a26e0b58862bc86d1ef
* :green_heart: [Build #149792](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/) succeeded 760e7028c2dfa3a7fd14221d7d7f0722fe6f2858
* :green_heart: [Build #149692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/) succeeded 45e46d1e526e940f75bf354dfb41fd94939be1c8
* :broken_heart: [Build #149510](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/) failed 57d7cad693b86d2b7522feb2fb8917282994b2ec
* :broken_heart: [Build #149445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/) failed fcc34f0df1ffd36ff0952dc0d288c9801a4c0274

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149445/","number":"149445","commit":"fcc34f0df1ffd36ff0952dc0d288c9801a4c0274"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149510/","number":"149510","commit":"57d7cad693b86d2b7522feb2fb8917282994b2ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149692/","number":"149692","commit":"45e46d1e526e940f75bf354dfb41fd94939be1c8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/","number":"149792","commit":"760e7028c2dfa3a7fd14221d7d7f0722fe6f2858"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/","number":"149824","commit":"c8e03e1512325e3a9f988a26e0b58862bc86d1ef"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/","number":"150175","commit":"a582d0ed6b30ceab32f85de1f92932392ba00496"}]}
PIPELINE-->

stratoula commented:
@jloleysens everything works perfect except from one case.
I create an area visualization from a saved search ([eCommerce] Orders). I don't save it and hit Share, report is generated with errors. If I save it, it is generated correctly.

Except from that everything looks awesome! 🎉

jloleysens commented:
@stratoula I was able to reproduce an issue, but this looks like it might be something pre-existing in visualizations:

1. Create an area vis using the `[eCommerce] Orders` saved search
2. Add an x-axis with date histogram agg.
3. Press "Update" to see the date histogram appear in the visualization and have the URL update
3. Refresh the page (without saving) and you get an error that says: `Trying to initialize aggs without index pattern`

It looks like this is because the `indexPattern` is not captured in the URL (so not passed to the vis when it is initialized). I've worked around if for now for report generation by reading the `indexPattern` from `visInstance.savedSearch.searchSource.getField('index').id` to the locator, but I suspect a more robust solution will be to add this to the URL as we do with `searchSourceId`.

If you agree I think we should do this on another PR. WDYT?

Otherwise, let me know whether the issue you running into has been addressed 👍🏻

tsullivan commented on src/plugins/visualize/public/application/utils/get_top_nav_config.tsx:
```diff
@@ -279,6 +283,23 @@ export const getTopNavConfig = (
+            type: currentState.vis.type,
+            uiState: currentState.uiState,
+            query: currentState.query,
+            vis: currentState.vis,
```
Would the Visualize app be able to query the saved objects service for this object using the `visId`?

Just curious for now. It could be something to optimize in the future.

jloleysens replied:
That's a really good point 👍🏻  I'll open an issue for tracking that after we merge this PR.

tsullivan commented on src/plugins/visualize/common/locator.ts:
```diff
@@ -0,0 +1,139 @@
+   *
+   * @note This is required to navigate to "create" page (i.e., when no `visId` has been provided).
+   */
+  type?: string;
```
It looks like `type` is also available in `PureVisState` under the `vis` field.  Not sure if the references that need this could use that field instead.

jloleysens replied:
Good point, let's remove it from the locator params 👌🏻

tsullivan reviewed (approved):
Code changes LGTM. Thanks so much! This is a huge user experience win.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150886/)
* Commit: 6ac1ed552c58433d8141d89cc5628531e884faa1
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150886/execution/node/428/log)

### History
* :green_heart: [Build #150514](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150514/) succeeded fc95193a5142e080e26eddf8fa0608520bec52f0
* :green_heart: [Build #150200](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150200/) succeeded 53d92d79ae390f55d99155921c33476177aa3c2f
* :green_heart: [Build #150175](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/) succeeded a582d0ed6b30ceab32f85de1f92932392ba00496
* :green_heart: [Build #149824](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/) succeeded c8e03e1512325e3a9f988a26e0b58862bc86d1ef
* :green_heart: [Build #149792](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/) succeeded 760e7028c2dfa3a7fd14221d7d7f0722fe6f2858

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149792/","number":"149792","commit":"760e7028c2dfa3a7fd14221d7d7f0722fe6f2858"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/","number":"149824","commit":"c8e03e1512325e3a9f988a26e0b58862bc86d1ef"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/","number":"150175","commit":"a582d0ed6b30ceab32f85de1f92932392ba00496"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150200/","number":"150200","commit":"53d92d79ae390f55d99155921c33476177aa3c2f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150514/","number":"150514","commit":"fc95193a5142e080e26eddf8fa0608520bec52f0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150886/","number":"150886","commit":"6ac1ed552c58433d8141d89cc5628531e884faa1"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150851/)
* Commit: 7fe9e6057121d3535058ed2dd908cbf1344255e2
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110206/7fe9e6057121d3535058ed2dd908cbf1344255e2)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/7fe9e6057121d3535058ed2dd908cbf1344255e2) | diff |
| --- | --- | --- | --- |
| `visualize` | 81 | 83 | +2 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/7fe9e6057121d3535058ed2dd908cbf1344255e2) | diff |
| --- | --- | --- | --- |
| `visualizations` | 261 | 253 | -8 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/7fe9e6057121d3535058ed2dd908cbf1344255e2) | diff |
| --- | --- | --- | --- |
| `visualize` | 95.4KB | 89.1KB | -6.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/7fe9e6057121d3535058ed2dd908cbf1344255e2) | diff |
| --- | --- | --- | --- |
| `visualize` | 27.8KB | 39.6KB | +11.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/7fe9e6057121d3535058ed2dd908cbf1344255e2) | diff |
| --- | --- | --- | --- |
| `visualizations` | 279 | 271 | -8 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/7fe9e6057121d3535058ed2dd908cbf1344255e2) | diff |
| --- | --- | --- | --- |
| `visualize` | 78 | 93 | +15 |


</details>

### History
* :broken_heart: [Build #150886](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150886/) failed 6ac1ed552c58433d8141d89cc5628531e884faa1
* :green_heart: [Build #150514](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150514/) succeeded fc95193a5142e080e26eddf8fa0608520bec52f0
* :green_heart: [Build #150200](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150200/) succeeded 53d92d79ae390f55d99155921c33476177aa3c2f
* :green_heart: [Build #150175](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/) succeeded a582d0ed6b30ceab32f85de1f92932392ba00496
* :green_heart: [Build #149824](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/) succeeded c8e03e1512325e3a9f988a26e0b58862bc86d1ef

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149824/","number":"149824","commit":"c8e03e1512325e3a9f988a26e0b58862bc86d1ef"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/","number":"150175","commit":"a582d0ed6b30ceab32f85de1f92932392ba00496"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150200/","number":"150200","commit":"53d92d79ae390f55d99155921c33476177aa3c2f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150514/","number":"150514","commit":"fc95193a5142e080e26eddf8fa0608520bec52f0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150886/","number":"150886","commit":"6ac1ed552c58433d8141d89cc5628531e884faa1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150851/","number":"150851","commit":"7fe9e6057121d3535058ed2dd908cbf1344255e2"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150908/)
* Commit: d1ad262e3dd0afb6266b8e846db2de4fa86586be
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110206/d1ad262e3dd0afb6266b8e846db2de4fa86586be)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/d1ad262e3dd0afb6266b8e846db2de4fa86586be) | diff |
| --- | --- | --- | --- |
| `visualize` | 81 | 83 | +2 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/d1ad262e3dd0afb6266b8e846db2de4fa86586be) | diff |
| --- | --- | --- | --- |
| `visualizations` | 261 | 253 | -8 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/d1ad262e3dd0afb6266b8e846db2de4fa86586be) | diff |
| --- | --- | --- | --- |
| `visualize` | 95.4KB | 89.0KB | -6.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/d1ad262e3dd0afb6266b8e846db2de4fa86586be) | diff |
| --- | --- | --- | --- |
| `visualize` | 27.8KB | 39.6KB | +11.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/d1ad262e3dd0afb6266b8e846db2de4fa86586be) | diff |
| --- | --- | --- | --- |
| `visualizations` | 279 | 271 | -8 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/33c124b9d8f09f07ae236098424de08c4ae3827e) | [after](https://github.com/elastic/kibana/commit/d1ad262e3dd0afb6266b8e846db2de4fa86586be) | diff |
| --- | --- | --- | --- |
| `visualize` | 78 | 93 | +15 |


</details>

### History
* :broken_heart: [Build #150886](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150886/) failed 6ac1ed552c58433d8141d89cc5628531e884faa1
* :green_heart: [Build #150851](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150851/) succeeded 7fe9e6057121d3535058ed2dd908cbf1344255e2
* :green_heart: [Build #150514](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150514/) succeeded fc95193a5142e080e26eddf8fa0608520bec52f0
* :green_heart: [Build #150200](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150200/) succeeded 53d92d79ae390f55d99155921c33476177aa3c2f
* :green_heart: [Build #150175](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/) succeeded a582d0ed6b30ceab32f85de1f92932392ba00496

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150175/","number":"150175","commit":"a582d0ed6b30ceab32f85de1f92932392ba00496"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150200/","number":"150200","commit":"53d92d79ae390f55d99155921c33476177aa3c2f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150514/","number":"150514","commit":"fc95193a5142e080e26eddf8fa0608520bec52f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150851/","number":"150851","commit":"7fe9e6057121d3535058ed2dd908cbf1344255e2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150886/","number":"150886","commit":"6ac1ed552c58433d8141d89cc5628531e884faa1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150908/","number":"150908","commit":"d1ad262e3dd0afb6266b8e846db2de4fa86586be"}]}
PIPELINE-->
